firefox_ui_tests/functional/sessionstore/test_restore_windows_after_restart.py

```diff
@@ -0,0 +1,94 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import Wait
+
+import time
+
+from firefox_puppeteer.testcases import FirefoxTestCase
+
+
+class TestRestoreWindowsAfterRestart(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        # Each list element represents a window of tabs loaded at
+        # some testing URL
+        self.test_windows = [
+          # Window 1
+          [self.marionette.absolute_url('layout/mozilla.html'),
+           self.marionette.absolute_url('layout/mozilla_community.html')],
+
+          # Window 2
+          [self.marionette.absolute_url('layout/mozilla_governance.html'),
+           self.marionette.absolute_url('layout/mozilla_grants.html')],
+
+          # Window 3
+          [self.marionette.absolute_url('layout/mozilla.html')],
+        ]
+
+        self.private_windows = [
+          [self.marionette.absolute_url('layout/mozilla_mission.html'),
+           self.marionette.absolute_url('layout/mozilla_organizations.html')],
+
+          [self.marionette.absolute_url('layout/mozilla_projects.html'),
+           self.marionette.absolute_url('layout/mozilla_mission.html')],
+        ]
+
+        # Set browser to restore previous session
+        self.prefs.set_pref('browser.startup.page', 3)
+
+
+    def tearDown(self):
+        try:
+            self.windows.close_all([self.browser.tabbar.tabs[0]])
```

Comment:
No, with `self.browser` I meant the entry in the exclusion list (parameter) of the function call. As of now you are  trying to close all windows, except the first tab of the browser window. By using that Marionette will also close the initial browser window, and will hang until the global timeout.